File: services/api/app/schemas/asset.py
"""Asset schemas for serialization."""

from datetime import date
from pydantic import BaseModel, ConfigDict


class AssetResponse(BaseModel):
    """Schema representing a fixed infrastructure asset."""

    asset_id: str
    section_id: str | None = None
    station_code: str | None = None
    department: str
    asset_type: str
    failure_risk_score: float | None = None
    criticality_index: float | None = None
    last_maintained_date: date | None = None
    source_type: str | None = None

    model_config = ConfigDict(from_attributes=True)


class AssetDetailResponse(AssetResponse):
    """Detailed schema representing an asset."""

    pass

File: services/web/src/lib/types/asset.ts
export interface Asset {
  asset_id: string;
  section_id: string | null;
  station_code: string | null;
  department: string;
  asset_type: string;
  failure_risk_score: number | null;
  criticality_index: number | null;
  last_maintained_date: string | null;
  source_type: string | null;
}
